### ISSUE TITLE
cpp11 code builds but I can't fit a model

### ISSUE THREAD

**Issue by pachadotdev** [pachadotdev/fixest2#21]
I still can't fit a model, but there's huge progress here

so far I see this after installing from the cpp11 branch

```
cpp11::cpp_register()
devtools::install()
fixest::feols(mpg ~ wt, data = mtcars)
Error: r_vector
``` 

here's the full error msg
```
3. cpp_sparse_products(X, w, y, correct_0w, nthreads) at ESTIMATION_FUNS.R#1792
2. ols_fit(y_demean, X_demean, weights, correct_0w, collin.tol, 
    nthreads, xwx, xwy, only.coef = only.coef) at ESTIMATION_FUNS.R#1525
1. feols(mpg ~ wt, data = mtcars)
```

**Comment by mpadge** [comment]
@pachadotdev I was going to have a look for you, but can't install because the `.onLoad` fails to find `setFixest_nthreads()`. That function is not included anywhere in the code, but is in `MiscFuns.R` in the master branch. Copying that across generates another "could not find function `renvir_get`" error. Could you please first make sure this error is reproducible in a fresh clone of the repo?

**Comment by thisisnic** [comment]
@pachadotdev Is this running in VSCode?  Could you try it in RStudio and see if there's a more detailed error message? I remember last time there seemed to be a difference in level of detail in the error messages between the two.

**Comment by pachadotdev** [comment]
thanks a lot @mpadge @thisisnic 
I compile my code by running https://github.com/pachadotdev/fixest/blob/cpp11/vscode-install.r directly in there
it builds but then I can't get the estimated coefficients for any model because it fails with "r_vector"
I created a new branch with some refactoring on the R side to provide smaller scripts https://github.com/pachadotdev/fixest/tree/cpp11-some-r-refactoring

**Comment by mpadge** [comment]
@pachadotdev but that install still won't work on a fresh clone. Can you please try a fresh clone of repo -> devtools install/load_all/whatever, and you'll see it won't even install yet

**Comment by pachadotdev** [comment]
thanks @mpadge 

I tested and it works here :(

this is my full log and steps:

```bash
github $ git clone --depth 1 --branch cpp11-some-r-refactoring git@github.com:pachadotdev/fixest.git fixest2
Cloning into 'fixest2'...
remote: Enumerating objects: 573, done.
remote: Counting objects: 100% (573/573), done.
remote: Compressing objects: 100% (501/501), done.
remote: Total 573 (delta 137), reused 306 (delta 61), pack-reused 0
Receiving objects: 100% (573/573), 220.24 MiB | 7.18 MiB/s, done.
Resolving deltas: 100% (137/137), done.
Updating files: 100% (581/581), done.
github $ cd fixest2
fixest2 (cpp11-some-r-refactoring) $ Rscript vscode-install.r 

Attaching package: ‘testthat’

The following object is masked from ‘package:devtools’:

    test_file


✔  checking for file ‘/home/pacha/github/fixest2/DESCRIPTION’ ...
─  preparing ‘fixest’:
─  checking DESCRIPTION meta-information ...Warning in person1(given = given[[i]], family = family[[i]], middle = middle[[i]],  :
     It is recommended to use ‘given’ instead of ‘middle’.
    OK
─  cleaning src
   Warning in person1(given = given[[i]], family = family[[i]], middle = middle[[i]],  :
     It is recommended to use ‘given’ instead of ‘middle’.
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘fixest_2022.05.25.tar.gz’
   
Running /usr/lib/R/bin/R CMD INSTALL /tmp/Rtmp8QfFdN/fixest_2022.05.25.tar.gz \
  --install-tests 
* installing to library ‘/home/pacha/R/x86_64-pc-linux-gnu-library/4.2’
* installing *source* package ‘fixest’ ...
** using staged installation
Warning in person1(given = given[[i]], family = family[[i]], middle = middle[[i]],  :
  It is recommended to use ‘given’ instead of ‘middle’.
** libs
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 00_common.cpp -o 00_common.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 01_01_convergence_helpers.cpp -o 01_01_convergence_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 01_02_convergence_exp_family.cpp -o 01_02_convergence_exp_family.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 01_03_convergence_clusters.cpp -o 01_03_convergence_clusters.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 01_04_convergence_derivatives.cpp -o 01_04_convergence_derivatives.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 01_05_convergence.cpp -o 01_05_convergence.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 02_01_demeaning_helpers.cpp -o 02_01_demeaning_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 02_02_linalg.cpp -o 02_02_linalg.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 02_03_fixed_effects.cpp -o 02_03_fixed_effects.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 02_04_demeaning.cpp -o 02_04_demeaning.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 03_01_dot_square_bracket.cpp -o 03_01_dot_square_bracket.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 04_01_linear_model_helpers.cpp -o 04_01_linear_model_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 04_02_linear_model_cholesky.cpp -o 04_02_linear_model_cholesky.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 04_03_linear_model_mmult.cpp -o 04_03_linear_model_mmult.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 04_04_linear_model_ivs.cpp -o 04_04_linear_model_ivs.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_01_misc_helpers.cpp -o 05_01_misc_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_02_misc_linalg.cpp -o 05_02_misc_linalg.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_03_misc_exp_family.cpp -o 05_03_misc_exp_family.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_04_misc_derivatives.cpp -o 05_04_misc_derivatives.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_05_misc_tables.cpp -o 05_05_misc_tables.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_06_misc_tapplies.cpp -o 05_06_misc_tapplies.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 05_07_misc_lag.cpp -o 05_07_misc_lag.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 06_01_quf_helpers.cpp -o 06_01_quf_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 06_02_quf_refactor.cpp -o 06_02_quf_refactor.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 06_03_quf_tables.cpp -o 06_03_quf_tables.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 07_01_parallel_helpers.cpp -o 07_01_parallel_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 07_02_parallel_math.cpp -o 07_02_parallel_math.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 07_03_parallel_exp_family.cpp -o 07_03_parallel_exp_family.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 07_04_parallel_linalg.cpp -o 07_04_parallel_linalg.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 08_01_strings_helpers.cpp -o 08_01_strings_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 09_01_vcov_helpers.cpp -o 09_01_vcov_helpers.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 09_02_vcov_dwdk.cpp -o 09_02_vcov_dwdk.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c 09_03_vcov_conley.cpp -o 09_03_vcov_conley.o
g++ -std=gnu++17 -I"/usr/share/R/include" -DNDEBUG  -I'/home/pacha/R/x86_64-pc-linux-gnu-library/4.2/cpp11/include'   -fopenmp -Wall -pedantic -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-zYgbYq/r-base-4.2.1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c cpp11.cpp -o cpp11.o
g++ -std=gnu++17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o fixest.so 00_common.o 01_01_convergence_helpers.o 01_02_convergence_exp_family.o 01_03_convergence_clusters.o 01_04_convergence_derivatives.o 01_05_convergence.o 02_01_demeaning_helpers.o 02_02_linalg.o 02_03_fixed_effects.o 02_04_demeaning.o 03_01_dot_square_bracket.o 04_01_linear_model_helpers.o 04_02_linear_model_cholesky.o 04_03_linear_model_mmult.o 04_04_linear_model_ivs.o 05_01_misc_helpers.o 05_02_misc_linalg.o 05_03_misc_exp_family.o 05_04_misc_derivatives.o 05_05_misc_tables.o 05_06_misc_tapplies.o 05_07_misc_lag.o 06_01_quf_helpers.o 06_02_quf_refactor.o 06_03_quf_tables.o 07_01_parallel_helpers.o 07_02_parallel_math.o 07_03_parallel_exp_family.o 07_04_parallel_linalg.o 08_01_strings_helpers.o 09_01_vcov_helpers.o 09_02_vcov_dwdk.o 09_03_vcov_conley.o cpp11.o -fopenmp -L/usr/lib/R/lib -lR
installing to /home/pacha/R/x86_64-pc-linux-gnu-library/4.2/00LOCK-fixest/00new/fixest/libs
** R
** data
*** moving datasets to lazyload DB
** demo
** tests
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
FIXEST
ℹ This is a fork of the original project created by Laurent Berge.
ℹ Visit https://buymeacoffee.com/pacha if you decide to donate and contribute to improve this project.
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
FIXEST
ℹ This is a fork of the original project created by Laurent Berge.
ℹ Visit https://buymeacoffee.com/pacha if you decide to donate and contribute to improve this project.
** testing if installed package keeps a record of temporary installation path
* DONE (fixest)
fixest2 (cpp11-some-r-refactoring) $ R

R version 4.2.1 (2022-06-23) -- "Funny-Looking Kid"
Copyright (C) 2022 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.


Attaching package: ‘testthat’

The following object is masked from ‘package:devtools’:

    test_file

> library(fixest)
FIXEST
ℹ This is a fork of the original project created by Laurent Berge.
ℹ Visit https://buymeacoffee.com/pacha if you decide to donate and contribute to improve this project.
> feols(mpg ~ wt, data = mtcars)
Error: r_vector
> 
```

**Comment by mpadge** [comment]
cool, thanks Pacha, all works on the cpp11-refactor branch. I'll dig a bit further this coming week and get back to you

**Comment by pachadotdev** [comment]
> 

hi @mpadge 
sorry to insist on this, could you have a look? I just came back to the issue now that tabulizer works again

**Comment by mpadge** [comment]
It's caused by this line: https://github.com/pachadotdev/fixest2/blob/ce2b0d07ef9d95c0f7cf6e58bb80a00e8ebeb72e/src/04_03_linear_model_mmult.cpp#L220 Standard debugging tools will enable you to find that -  like [the tools I describe in this blog post](https://mpadge.github.io/blog/blog012.html). My understanding of `cpp11` is that tou can only refer to named list items for extraction; created objects can only be inserted by indexing, with names only able to be imposed on the R side once the list has been returned from C++ code.

**Comment by pachadotdev** [comment]
> It's caused by this line:
> 
> https://github.com/pachadotdev/fixest2/blob/ce2b0d07ef9d95c0f7cf6e58bb80a00e8ebeb72e/src/04_03_linear_model_mmult.cpp#L220
> Standard debugging tools will enable you to find that - like [the tools I describe in this blog post](https://mpadge.github.io/blog/blog012.html). My understanding of `cpp11` is that tou can only refer to named list items for extraction; created objects can only be inserted by indexing, with names only able to be imposed on the R side once the list has been returned from C++ code.

thanks a lot
my approach to C++ is quite elemental, just notepad and that's it

I couldn't get `PKG_CPPFLAGS=UDEBUG -g` to work so I consistently implemented a push back for that script https://github.com/pachadotdev/fixest2/commit/fc253cfb45769707a1df1d0ac820117617a8727c

I'll keep trying, thanks a lot!

**Comment by pachadotdev** [comment]
@mpadge I suggested a change to your post, a 1 million dollar 1-character modification https://github.com/mpadge/mpadge.github.io/compare/source...pachadotdev:mpadge.github.io:patch-1

advanced PhD studies were involved to detect it :O

**Comment by pachadotdev** [comment]
fixed in efb8d422c9ea57ea71b851112c02be5522d3500b
works for lm without offset at the moment
